python/cudf/cudf/core/column/categorical.py

```diff
@@ -90,6 +90,10 @@ def set_categories(self, new_categories, **kwargs):
         ordered = kwargs.get("ordered", self.ordered)
         rename = kwargs.pop("rename", False)
         new_categories = column.as_column(new_categories)
+
+        if hasattr(new_categories, "categories"):
```

Comment:
Given `new_categories` is guaranteed to be a `Column` object at this point, should we just do `isinstance(new_categories, CategoricalColumn)` instead? It would be cheaper than a `hasattr` check.